### PR TITLE
Set ROM degree

### PR DESCRIPTION
**Pull request by Schaeff**
When creating the ROM machine, give it a degree:
- if the VM has a degree, pass the same degree to its ROM machine (this is required to support the monolithic case where all degrees must match)
- otherwise, pass the length of the ROM

### CODE REVIEW

**Comment by georgwiese** [comment]
This branch is already working well with #1676, but I needed to make a few small changes to fix the tests: 9f0484db1ace3dca2e83184cbe228dfe9d8bb044

File: test_data/asm/vm_args_relative_path.asm [hunk 1]
```diff
@@ -1,4 +1,9 @@
-machine Main with degree: 16 {
+// we cannot have `N` at the top level because of how shorts paths are displayed with `.` instead of `::`
```
Comment (leonardoalt) [inline test_data/asm/vm_args_relative_path.asm:1]: shorts paths?

File: pipeline/tests/asm.rs [hunk 2]
```diff
@@ -179,7 +179,8 @@ fn vm_to_vm() {
 #[test]
 fn vm_to_vm_dynamic_trace_length() {
     let f = "asm/vm_to_vm_dynamic_trace_length.asm";
-    regular_test(f, &[]);
+    let pipeline = make_simple_prepared_pipeline(f);
+    test_halo2_with_backend_variant(pipeline.clone(), BackendVariant::Composite);
```
Comment (leonardoalt) [inline pipeline/tests/asm.rs:183]: why halo2 only?

**Comment by leonardoalt** [comment]
> otherwise, pass the smallest power of two which fits the ROM

Not sure if important, but the implementation in this PR doesn't do next power of 2, but rather the size of the code itself.

**Comment by Schaeff** [comment]
Good point, I was referring to https://github.com/powdr-labs/powdr/commit/9f0484db1ace3dca2e83184cbe228dfe9d8bb044 thinking it was pushed here but it's not. Correcting the PR description.

File: test_data/asm/vm_args_relative_path.asm (hunk 1 above)
Comment (Schaeff) [reply test_data/asm/vm_args_relative_path.asm:1]: This is an existing issue where `a::b` gets printed as `a.b` to support both PIL and asm. The workaround here is to have a longer path.

File: pipeline/tests/asm.rs (hunk 2 above)
Comment (Schaeff) [reply pipeline/tests/asm.rs:183]: Oh, actually it should work on all composite backends.

File: test_data/asm/vm_args_relative_path.asm (hunk 1 above)
Comment (leonardoalt) [reply test_data/asm/vm_args_relative_path.asm:1]: Ah ok. "Short path" then

File: asm-to-pil/src/vm_to_constrained.rs [hunk 3]
```diff
@@ -278,9 +280,18 @@ impl<T: FieldElement> VMConverter<T> {
             input.pil.extend(self.pil);
         }
 
+        // This is hacky: in the absence of proof objects, we want to support both monolithic proofs and composite proofs.
+        // In the monolithic case, all degrees must be the same, so we align the degree of the rom to that of the vm.
+        // In the composite case, we set the minimum degree for the rom, which is the number of lines in the code.
+        let rom_degree = input
+            .degree
+            .clone()
+            .unwrap_or_else(|| Expression::from(self.code_lines.len() as u32));
```
Comment (georgwiese) [inline asm-to-pil/src/vm_to_constrained.rs:289]:
```suggestion
            .unwrap_or_else(|| Expression::from(self.code_lines.len().next_power_of_two() as u32));
```

Comment (georgwiese) [reply asm-to-pil/src/vm_to_constrained.rs:289]: Or you can cherry-pick 2773609db21f947bd675c173977ea690d8ce0be7 to adjust the tests as well.

Comment (Schaeff) [reply asm-to-pil/src/vm_to_constrained.rs]: done

File: test_data/asm/vm_args_relative_path.asm (hunk 1 above)
Comment (Schaeff) [reply test_data/asm/vm_args_relative_path.asm:1]: done

File: pipeline/tests/asm.rs (hunk 2 above)
Comment (Schaeff) [reply pipeline/tests/asm.rs:183]: done

**Review by georgwiese — DISMISSED** [review]
🚀

**Review by georgwiese — APPROVED** [review]

**Pull request merged**